algorithms/unit_tests/TestStdAlgorithmsCompileOnly.cpp

```diff
@@ -0,0 +1,655 @@
+//@HEADER
+*/
+
+#include <std_algorithms/Kokkos_BeginEnd.hpp>
+#include <Kokkos_StdAlgorithms.hpp>
+
+namespace Test {
+namespace stdalgos {
+namespace compileonly {
+
+template <class ValueType>
+struct TrivialUnaryFunctor {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const ValueType a) const { return a; }
+};
+
+template <class ValueType>
+struct TrivialBinaryFunctor {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const ValueType &a, const ValueType &b) const {
+    return (a + b);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const volatile ValueType &a,
+                       const volatile ValueType &b) const {
+    return (a + b);
+  }
+};
+
+template <class ValueType>
+struct TrivialUnaryPredicate {
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(const ValueType val) const {
+    (void)val;
+    return true;
+  }
+};
+
+template <class ValueType>
+struct TimesTwoFunctor {
+  KOKKOS_INLINE_FUNCTION
+  void operator()(ValueType &val) const { val *= (ValueType)2; }
+};
+
+template <class ValueType>
+struct TrivialComparator {
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(const ValueType &a, const ValueType &b) const {
+    return a > b;
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(const volatile ValueType &a,
+                  const volatile ValueType &b) const {
+    return a > b;
+  }
+};
+
+template <class ValueType>
+struct TrivialGenerator {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()() const { return ValueType{}; }
+};
+
+template <class ValueType>
+struct TrivialReduceJoinFunctor {
+  KOKKOS_FUNCTION
+  ValueType operator()(const ValueType &a, const ValueType &b) const {
+    return a + b;
+  }
+
+  KOKKOS_FUNCTION
+  ValueType operator()(const volatile ValueType &a,
+                       const volatile ValueType &b) const {
+    return a + b;
+  }
+};
+
+template <class ValueType>
+struct TrivialTransformReduceUnnaryTransformer {
+  KOKKOS_FUNCTION
+  ValueType operator()(const ValueType &a) const { return a; }
+};
+
+template <class ValueType>
+struct TrivialTransformReduceBinaryTransformer {
+  KOKKOS_FUNCTION
+  ValueType operator()(const ValueType &a, const ValueType &b) const {
+    return (a * b);
+  }
+};
+
+// put all code here and don't call from main
+// so that even if one runs the executable,
+// nothing is run anyway
+void all() {
+  namespace KE = Kokkos::Experimental;
+  Kokkos::DefaultExecutionSpace execution_space;
+
+  {
+    using T = double;
+    Kokkos::View<T *> in1("in1", 10);
+    auto r = KE::find("shouldwork", execution_space, in1, 5.);
+    (void)r;
+  }
+
+  {
+    using T = double;
+    Kokkos::View<T *> in1("in1", 10);
+    auto r = KE::find_if("shouldwork", execution_space, in1,
+                         TrivialUnaryPredicate<T>());
+    (void)r;
+  }
+
+  {
+    using T = double;
+    Kokkos::View<T *> in1("in1", 10);
+    auto r = KE::find_if_not("shouldwork", execution_space, in1,
+                             TrivialUnaryPredicate<T>());
+    (void)r;
+  }
```

Comment:
I added only tests for overloads accepting view since I thought those were the ones potentially causing issues.  
I will revise to call all of them like you suggested. 

Btw, the only one (at least it seems so) triggering ambiguous call was `transform` :)